### PR TITLE
Add definitions for Ruby 2.7.8, 3.0.6, 3.1.4, 3.2.2

### PR DESCRIPTION
**Pull request by jas14**
See security/release announcements for:

* [2.7.8](https://www.ruby-lang.org/en/news/2023/03/30/ruby-2-7-8-released/)
* [3.0.6](https://www.ruby-lang.org/en/news/2023/03/30/ruby-3-0-6-released/)
* [3.1.4](https://www.ruby-lang.org/en/news/2023/03/30/ruby-3-1-4-released/)
* [3.2.2](https://www.ruby-lang.org/en/news/2023/03/30/ruby-3-2-2-released/)

### CODE REVIEW

File: share/ruby-build/3.1.4 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+install_package "openssl-3.0.8" "https://www.openssl.org/source/openssl-3.0.8.tar.gz#6c13d2bf38fdf31eac3ce2a347073673f5d63263398f1f69d0df4a41253e4b3e" openssl --if needs_openssl_102_300
```
Comment (spickermann) [inline share/ruby-build/3.1.4:1]:
```suggestion
install_package "openssl-3.1.0" "https://www.openssl.org/source/openssl-3.1.0.tar.gz#aaa925ad9828745c4cad9d9efeb273deca820f2cdcf2c3ac7d7c1212b7c497b4" openssl --if needs_openssl_102_300
```

File: share/ruby-build/3.2.2 [hunk 2]
```diff
@@ -0,0 +1,2 @@
+install_package "openssl-3.0.8" "https://www.openssl.org/source/openssl-3.0.8.tar.gz#6c13d2bf38fdf31eac3ce2a347073673f5d63263398f1f69d0df4a41253e4b3e" openssl --if needs_openssl_102_300
```
Comment (spickermann) [inline share/ruby-build/3.2.2:1]:
```suggestion
install_package "openssl-3.1.0" "https://www.openssl.org/source/openssl-3.1.0.tar.gz#aaa925ad9828745c4cad9d9efeb273deca820f2cdcf2c3ac7d7c1212b7c497b4" openssl --if needs_openssl_102_300
```

**Review by spickermann — COMMENTED** [review]
I suggest updating to the [latest version OpenSSL](https://www.openssl.org/source/) too.

**Comment by nafu** [comment]
Thank you for the support quickly!

**Comment by SeanSith** [comment]
@mislav Thanks for knocking this out! That was super fast! Is the tag for the release waiting on #2175 ? I was going to kick a PR over to asdf-vm/asdf-ruby.

**Pull request merged**